apps/src/templates/teacherNavigation/ElementOrEmptyPage.tsx

```diff
@@ -27,82 +33,155 @@ const ElementOrEmptyPage: React.FC<ElementOrEmptyPageProps> = ({
   showNoStudents,
   showNoCurriculumAssigned,
   showNoUnitAssigned,
+  showNoLessonMaterialsForLegacyCourses,
+  showNoCalendarForLegacyCourses,
+  showNoLessonMaterialsForLesson,
+  showNoCalendarForThisUnit,
+  isOnCalendarPage,
   courseName,
   element,
 }) => {
   const isLoadingSectionData = useAppSelector(
     state => state.teacherSections.isLoadingSectionData
   );
 
-  const textDescription = () => {
-    if (showNoStudents) {
-      return i18n.emptySectionDescription();
-    } else if (showNoCurriculumAssigned) {
-      return i18n.noCurriculumAssigned();
-    } else {
-      return i18n.noUnitAssigned({courseName: courseName});
-    }
-  };
+  const lessonMaterialsOrCalendarPage = isOnCalendarPage
+    ? i18n.theCalendar()
+    : i18n.theLessonMaterials();
+
+  const navigate = useNavigate();
 
-  const displayedImage = () => {
-    if (showNoStudents) {
-      return <img src={emptyDesk} alt="empty desk" />;
-    } else if (showNoCurriculumAssigned) {
-      return <img src={blankScreen} alt="blank screen" />;
-    } else {
-      return <img src={TeacherDashboardEmptyState} alt={i18n.almostThere()} />;
-    }
+  const navigateToCoursePage = () => {
```

Comment:
Not necessarily for this PR, but we should change all the buttons that link to a new page using `onClick`. This is not an accessible pattern because it does not list the URL on hover. Instead we should use href on the button or a react-router-dom NavLink.